### DISCUSSION TITLE
List available api routes

### DISCUSSION THREAD

**Discussion by GeoSot** [binaryk/laravel-restify#430] [Q&A] [answered]
I had created some Repositories (as the examples) and was trying to find out a way to list my api available routes. 
In some of them I had created some custom routes too.

So I did an `artisan route:list` and surprisingly I got only the custom routes and not all the other 'out of the box' given routes. 
After some further checks, I was guided to this clever trick https://github.com/BinarCode/laravel-restify/blob/5.x/routes/api.php in where I found out that the routes are resolved really dynamically. 

I am ok witη it as it seems to work.
However, I want to use the package in a huge system, and I am sure it will be handy and consistent to have a way listing all the api routes somehow.

@binaryk I would appreciate any suggestions over it. ( the only thing I have in mind, includes registering routes separately for each repo, using a helper, in order to be read from native Route registrar .

(In case I am wrong on something, I can provide screenshots of my issue if you like)

PS: thanks again for your quick responses

**Comment by binaryk** [comment]
Hi @GeoSot ! Thanks for you suggestion. 

We're using it in many big production ready products, and honestly didn't need this so far. However, I see folks might be interested in it, so we'll implement it. [Here](https://github.com/BinarCode/laravel-restify/issues/444) is an issue around same Q. 

There are 2 ways to archive this:
- your idea, to iterate over all repositories and inject CRUD routes (not all from the `api.php`) into the Route Registrar in a booted Restify provider. Now, the downside of this, is devs will see a very long list of routes when running `route:list`, it'll be like: _6 x repositories count_
- the second idea is to build a custom command that could list Restify specific routes, and might accept the `repository` key, as well as `postman` export


Whichever do you consider more? 

Btw we updated the [documentation](https://restify.binarcode.com/quickstart) recently and released the `6.x` version of the library.

**Reply by GeoSot** [reply]
the first idea seems uglier but more consistent to a laravel project, as it follows the standard way to track project routes. As a plus, api custom routes will be in the same list as the defaults

The second idea, could work in combination with the first on, as yes, it is a very handy helper :wink: 

Thanks for answering :smiley:

**Comment by binaryk** [comment, marked as answer]
This was solved in 7.x. Thank you @GeoSot!
